wom/src/test/scala/wom/util/YamlUtilsSpec.scala

```diff
@@ -0,0 +1,156 @@
+package wom.util
+
+import com.typesafe.config.ConfigException.BadValue
+import com.typesafe.config.ConfigFactory
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.numeric.NonNegative
+import eu.timepit.refined.refineMV
+import io.circe.Json
+import net.ceedubs.ficus.Ficus._
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.{EitherValues, FlatSpec, Matchers}
+
+class YamlUtilsSpec extends FlatSpec with Matchers with TableDrivenPropertyChecks with EitherValues {
+
+  behavior of "YamlUtils"
+
+  private val illegalYamlTests = Table(
+    ("description", "yaml", "maxNodes", "exceptionMessage"),
+    (
+      // https://stackoverflow.com/a/16986339
+      "a stackoverflow yaml bomb",
+      "{a: &b [*b]}",
+      refineMV[NonNegative](10),
+      "Loop detected",
+    ),
+    (
+      // https://bitbucket.org/asomov/snakeyaml/wiki/Documentation#markdown-header-aliases
+      "a recursive yaml example from the snakeyaml wiki",
+      "&A [ *A ]",
+      refineMV[NonNegative](10),
+      "Loop detected",
+    ),
+    (
+      "a recursive yaml based on mappings back to the root",
+      """|a: &b
+         |  c: *b
+         |""".stripMargin,
+      refineMV[NonNegative](10),
+      "Loop detected",
+    ),
+    (
+      // https://bitbucket.org/asomov/snakeyaml-engine/src/41b3845/src/test/resources/recursive/recursive-set-1.yaml
+      "a recursive yaml set",
+      """|# Sets are represented as a
+         |# Mapping where each key is
+         |# associated with a null value
+         |--- !!set &key
+         |? Mark McGwire
+         |? Sammy Sosa
+         |? *key
+         |""".stripMargin,
+      refineMV[NonNegative](10),
+      "Loop detected",
+    ),
+    (
+      // https://en.wikipedia.org/w/index.php?title=Billion_laughs_attack&oldid=871224525#Variations
+      "a billion laughs when limited to 10,000 nodes",
+      """|a: &a ["lol","lol","lol","lol","lol","lol","lol","lol","lol"]
+         |b: &b [*a,*a,*a,*a,*a,*a,*a,*a,*a]
+         |c: &c [*b,*b,*b,*b,*b,*b,*b,*b,*b]
+         |d: &d [*c,*c,*c,*c,*c,*c,*c,*c,*c]
+         |e: &e [*d,*d,*d,*d,*d,*d,*d,*d,*d]
+         |f: &f [*e,*e,*e,*e,*e,*e,*e,*e,*e]
+         |g: &g [*f,*f,*f,*f,*f,*f,*f,*f,*f]
+         |h: &h [*g,*g,*g,*g,*g,*g,*g,*g,*g]
+         |i: &i [*h,*h,*h,*h,*h,*h,*h,*h,*h]
+         |""".stripMargin,
+      refineMV[NonNegative](10000),
+      "Loop detection halted at 10000 nodes",
+    ),
+    (
+      "an empty yaml mapping when limited to zero nodes",
+      "{}",
+      refineMV[NonNegative](0),
```

Comment:
TOL: to me, `0` in config is more often used to mean "turn off this setting" rather than "literally accept nothing at all".

Would I ever really want to set this to `0` for real?

Reply:
would you ever really want to turn it off?